### ISSUE TITLE
Windows: apply: Spawning child process: Exec format error

### ISSUE THREAD

**Issue by kantholtz** [org-roam/org-roam#2324]
### Description

Hello, first, thank you for the wonderful `org-roam` package I rely on daily! Unfortunately, I have a problem using org-roam on Windows (the same steps I describe here work fine under Linux). I am using version `v2.2.2-39-ge73807e`. It worked fine a few weeks ago, but I don't know the version of that. I have already reinstalled the package and the configuration from scratch.

I followed the official installation guidelines, installed `msys2`, updated all packages and installed `gcc` via pacman, and `(emacsql-sqlite-compile)` yields `t`. There is no database file in `.emacs.d` (there should be `"~/.emacs.d/org-roam.db"`? The stacktrace hints at that). Curiously, there is a running `emacsql-sqlite` process which complains when I want to exit Emacs:
```
emacsql-sqlite  -1      run      *emacsql-sqlite*         --           Main
```

Related issues I found are: 
- https://github.com/org-roam/org-roam/issues/1875
- https://github.com/org-roam/org-roam/issues/1855#issuecomment-920224285

#### Steps to Reproduce

The following commands trigger the same error: `(org-roam-db-sync)` and `(org-roam-node-find)`. The stacktrace can be found below.

#### Backtrace
```
Debugger entered--Lisp error: (file-error "Spawning child process" #("Exec format error" 0 17 (charset windows-1252)))
  #<subr make-process>(:name "emacsql-sqlite" :buffer #<buffer  *emacsql-sqlite*> :command ("c:/Users/micro/AppData/Roaming/.emacs.d/straight/b..." "c:/Users/micro/AppData/Roaming/.emacs.d/org-roam.d..."))
  apply(#<subr make-process> (:name "emacsql-sqlite" :buffer #<buffer  *emacsql-sqlite*> :command ("c:/Users/micro/AppData/Roaming/.emacs.d/straight/b..." "c:/Users/micro/AppData/Roaming/.emacs.d/org-roam.d...")))
  make-process--with-editor-process-filter(#<subr make-process> :name "emacsql-sqlite" :buffer #<buffer  *emacsql-sqlite*> :command ("c:/Users/micro/AppData/Roaming/.emacs.d/straight/b..." "c:/Users/micro/AppData/Roaming/.emacs.d/org-roam.d..."))
  apply(make-process--with-editor-process-filter #<subr make-process> (:name "emacsql-sqlite" :buffer #<buffer  *emacsql-sqlite*> :command ("c:/Users/micro/AppData/Roaming/.emacs.d/straight/b..." "c:/Users/micro/AppData/Roaming/.emacs.d/org-roam.d...")))
  make-process(:name "emacsql-sqlite" :buffer #<buffer  *emacsql-sqlite*> :command ("c:/Users/micro/AppData/Roaming/.emacs.d/straight/b..." "c:/Users/micro/AppData/Roaming/.emacs.d/org-roam.d..."))
  apply(make-process (:name "emacsql-sqlite" :buffer #<buffer  *emacsql-sqlite*> :command ("c:/Users/micro/AppData/Roaming/.emacs.d/straight/b..." "c:/Users/micro/AppData/Roaming/.emacs.d/org-roam.d...")))
  start-process("emacsql-sqlite" #<buffer  *emacsql-sqlite*> "c:/Users/micro/AppData/Roaming/.emacs.d/straight/b..." "c:/Users/micro/AppData/Roaming/.emacs.d/org-roam.d...")
  #f(compiled-function (connection &rest rest) #<bytecode 0x5879475>)(#<emacsql-sqlite-connection emacsql-sqlite-connection-5936ad0> (:file "~/.emacs.d/org-roam.db"))
  apply(#f(compiled-function (connection &rest rest) #<bytecode 0x5879475>) (#<emacsql-sqlite-connection emacsql-sqlite-connection-5936ad0> (:file "~/.emacs.d/org-roam.db")))
  #f(compiled-function (&rest args) #<bytecode 0x5936b0d>)(#<emacsql-sqlite-connection emacsql-sqlite-connection-5936ad0> (:file "~/.emacs.d/org-roam.db"))
  apply(#f(compiled-function (&rest args) #<bytecode 0x5936b0d>) #<emacsql-sqlite-connection emacsql-sqlite-connection-5936ad0> (:file "~/.emacs.d/org-roam.db"))
  initialize-instance(#<emacsql-sqlite-connection emacsql-sqlite-connection-5936ad0> (:file "~/.emacs.d/org-roam.db"))
  #f(compiled-function (class &rest slots) "Default constructor for CLASS `eieio-default-superclass'.\nSLOTS are the initialization slots used by `initialize-instance'.\nThis static method is called when an object is constructed.\nIt allocates the vector used to represent an EIEIO object, and then\ncalls `initialize-instance' on that object." #<bytecode 0x2e57fb1>)(emacsql-sqlite-connection :file "~/.emacs.d/org-roam.db")
  apply(#f(compiled-function (class &rest slots) "Default constructor for CLASS `eieio-default-superclass'.\nSLOTS are the initialization slots used by `initialize-instance'.\nThis static method is called when an object is constructed.\nIt allocates the vector used to represent an EIEIO object, and then\ncalls `initialize-instance' on that object." #<bytecode 0x2e57fb1>) emacsql-sqlite-connection (:file "~/.emacs.d/org-roam.db"))
  make-instance(emacsql-sqlite-connection :file "~/.emacs.d/org-roam.db")
  emacsql-sqlite("~/.emacs.d/org-roam.db")
  org-roam-db()
  org-roam-db-sync()
  org-roam-db-autosync-mode(1)
  org-roam-setup()
  (progn (org-roam-setup) t)
  (condition-case err (progn (org-roam-setup) t) ((debug error) (funcall use-package--warning33 :config err)))
  (condition-case-unless-debug err (progn (org-roam-setup) t) (error (funcall use-package--warning33 :config err)))
  (closure (t) nil (condition-case-unless-debug err (progn (org-roam-setup) t) (error (funcall use-package--warning33 :config err))))()
  eval-after-load-helper("c:/Users/micro/AppData/Roaming/.emacs.d/straight/b...")
  run-hook-with-args(eval-after-load-helper "c:/Users/micro/AppData/Roaming/.emacs.d/straight/b...")
  do-after-load-evaluation("c:/Users/micro/AppData/Roaming/.emacs.d/straight/b...")
  require(org-roam)
  byte-code("\300\301!\210\302\303\304\305\306DD\307\310\301\311\312&\7\210\302\313\304\305\314DD\315\310\301\311\305&\7\210\302\316\304\305\317DD\320\311\321\310\301&\7\210\302..." [require org-roam custom-declare-variable org-roam-node-display-template funcall function #f(compiled-function () #<bytecode 0x5866d95>) "Configures display formatting for Org-roam node.\nP..." :group :type (string function) org-roam-node-annotation-function #f(compiled-function () #<bytecode 0x5866da5>) "This function used to attach annotations for `org-..." org-roam-node-default-sort #f(compiled-function () #<bytecode 0x5866db5>) "Default sort order for Org-roam node completions." (choice (const :tag "none" nil) (const :tag "file-mtime" file-mtime) (const :tag "file-atime" file-atime)) org-roam-node-formatter #f(compiled-function () #<bytecode 0x5866dc5>) "The link description for node insertion.\nIf a func..." (string function) org-roam-node-template-prefixes #f(compiled-function () #<bytecode 0x5866dd5>) "Prefixes for each of the node's properties.\nThis i..." (alist) org-roam-ref-annotation-function #f(compiled-function () #<bytecode 0x5866de5>) "This function used to attach annotations for `org-..." (function) org-roam-ref-prompt-function #f(compiled-function () #<bytecode 0x5866df5>) "Function to prompt for ref strings in `org-roam-re..." org-roam-completion-everywhere #f(compiled-function () #<bytecode 0x5866e05>) "When non-nil, provide link completion matching out..." boolean org-roam-completion-functions #f(compiled-function () #<bytecode 0x5866e19>) "List of functions to be used with `completion-at-p..." hook org-roam-link-auto-replace #f(compiled-function () #<bytecode 0x5866e29>) "If non-nil, replace \"roam:\" links to existing node..." org-roam-extract-new-file-path #f(compiled-function () #<bytecode 0x5866e39>) "The file path template to use when a node is extra..." string] 8)
  autoload-do-load((autoload "org-roam-node" "Find and open an Org-roam node by its title or ali..." t nil) org-roam-node-find)
  command-execute(org-roam-node-find)
```

#### Expected Results

It should display the nodes ;)

#### Actual Results

The error I outlined above.

### Environment

- Emacs: GNU Emacs 27.2 (build 1, x86_64-w64-mingw32)
 of 2021-03-26
- Framework: N/A
- Org: Org mode version 9.6.1 (9.6.1-gaa2249 @ c:/Users/micro/AppData/Roaming/.emacs.d/straight/build/org/)
- Org-roam: v2.2.2-39-ge73807e
- sqlite-connector: sqlite

The OS is `Windows 11 Pro 22H2 (22621.1265)`

**Comment by kantholtz** [comment]
I've since updated to Emacs 28.2. I also provide two ways to reproduce the error - one using straight and use-package and one completely reduced to the bare minimum:

**straight/use-package**

- Emacs: GNU Emacs 28.2 (build 2, x86_64-w64-mingw32)
 of 2022-09-13
- Framework: N/A
- Org: Org mode version 9.6.1 (9.6.1-gaa2249 @ c:/Users/micro/AppData/Roaming/.emacs.d/straight/build/org/)
- Org-roam: v2.2.2-39-ge73807e
- sqlite-connector: sqlite


```elisp
(defvar bootstrap-version)
(let ((bootstrap-file
       (expand-file-name "straight/repos/straight.el/bootstrap.el" user-emacs-directory))
      (bootstrap-version 6))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/radian-software/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))

(setq package-gnupghome-dir nil)
(straight-use-package 'use-package)

(use-package org-roam
  :straight t
  :ensure t
  :config (org-roam-db-autosync-mode))
```

**raw**

- Emacs: GNU Emacs 28.2 (build 2, x86_64-w64-mingw32)
 of 2022-09-13
- Framework: 
- Org: Org mode version 9.5.5 (release_9.5.5 @ c:/Program Files/Emacs/emacs-28.2/share/emacs/28.2/lisp/org/)
- Org-roam: 2.2.2- sqlite-connector: sqlite

```elisp
(setq package-gnupghome-dir nil)
(require 'package)
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
(package-initialize)
(package-refresh-contents)
(package-install 'org-roam)
(org-roam-setup)
```

Both ways trigger the same error outlined above.

**Comment by soenderby** [comment]
I am currently having the same issue. 
One difference is that I have a populated database at `"~/.emacs.d/org-roam.db"`.

**Comment by kantholtz** [comment]
I made it work: I've completely removed all traces of MSYS2, reinstalled it and made sure to [use the `msys` environment](https://www.msys2.org/docs/environments/). Then I did the normal steps (`pacman -Syu`, close, `pacman -S gcc`). I don't know what changed, but I think I may have used a different environment before. However, given I understand the documentation correctly, this would've installed to a different prefix...
